cli/src/pcluster/models/cluster_resources.py

```diff
@@ -95,6 +96,91 @@ def _is_failed_wait(event):
         failure_event = next(filter(_is_failed_wait, stack_events), None)
         return failure_event.get("ResourceStatusReason") if failure_event else None
 
+    def get_cluster_creation_failure(self):
+        """Get error code and error reasons for cluster creation failure."""
+
+        @dataclass
```

Comment:
nitpick, this smells to me a bit, however I don't have real suggestions on how to make this cleaner. Just thinking of a separate file or a configuration file.

Have you thought of something else?

Reply:
Yeah, I have been thinking about putting this data class outside the function, but currently it is only used in this function so I leave it here for now.
I also think of using nametupled and dictionary but I think they are similar to dataclass.